src/redeem_token.cc

```diff
@@ -0,0 +1,419 @@
+
+void RedeemToken::CreateConfirmation(
+    const std::string& creative_instance_id,
+    const std::string& unblinded_token_base64) {
+  BLOG(INFO) << "CreateConfirmation";
+
+  // Create blinded token
+  auto payment_tokens = helper::Security::GenerateTokens(1);
+  auto payment_token = payment_tokens.front();
+  auto payment_token_base64 = payment_token.encode_base64();
+
+  auto blinded_payment_tokens = helper::Security::BlindTokens(payment_tokens);
+  auto blinded_payment_token = blinded_payment_tokens.front();
+  auto blinded_payment_token_base64 = blinded_payment_token.encode_base64();
+
+  // Generate confirmation id
+  auto confirmation_id = base::GenerateGUID();
+
+  // Create payload
+  base::DictionaryValue payload_dictionary;
+  payload_dictionary.SetKey("creativeInstanceId",
+      base::Value(creative_instance_id));
+  payload_dictionary.SetKey("payload",
+      base::Value(base::Value::Type::DICTIONARY));
+  payload_dictionary.SetKey("blindedPaymentToken",
+      base::Value(blinded_payment_token_base64));
+  payload_dictionary.SetKey("type", base::Value("landed"));
+
+  std::string payload_json;
+  base::JSONWriter::Write(payload_dictionary, &payload_json);
+
+  // Create credential
+  base::DictionaryValue credential_dictionary;
+
+  credential_dictionary.SetKey("payload", base::Value(payload_json));
+
+  auto unblinded_token = UnblindedToken::decode_base64(unblinded_token_base64);
+  auto unblinded_token_verification_key =
+      unblinded_token.derive_verification_key();
+  auto unblinded_token_verification_signature =
+      unblinded_token_verification_key.sign(payload_json);
+  auto unblinded_token_verification_signature_base64 =
+      unblinded_token_verification_signature.encode_base64();
+  credential_dictionary.SetKey("signature",
+      base::Value(unblinded_token_verification_signature_base64));
+
+  auto unblinded_token_preimage = unblinded_token.preimage();
+  auto unblinded_token_preimage_base64 =
+      unblinded_token_preimage.encode_base64();
+  credential_dictionary.SetKey("t",
+      base::Value(unblinded_token_preimage_base64));
+
+  std::string credential_json;
+  base::JSONWriter::Write(credential_dictionary, &credential_json);
+
+  std::vector<uint8_t> credential(credential_json.begin(),
+      credential_json.end());
+  std::string credential_base64 = helper::Security::GetBase64(credential);
+
+  std::string uri_encoded_credential =
+      net::EscapeQueryParamValue(credential_base64, true);
+
+  // Create request body
+  std::string body = payload_json;
+
+  // Create request headers
+  std::vector<std::string> headers = {};
+
+  auto accept_header = std::string("accept: ").append("application/json");
+  headers.push_back(accept_header);
+
+  // Create request content type
+  std::string content_type = "application/json";
+
+  // POST /v1/confirmation/{confirmation_id_}/{credential}
+  BLOG(INFO) << "POST /v1/confirmation/{confirmation_id_}/{credential}";
```

Comment:
Same comment as before, jsut checking if this is intentional (not printing the variables).